kai-client-ci/fixtures/jdtls/bin/jdtls.py

```diff
@@ -0,0 +1,114 @@
+###############################################################################
+# Copyright (c) 2022 Marc Schreiber and others.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0.
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+# Marc Schreiber - initial API and implementation
+###############################################################################
+import argparse
+from hashlib import sha1
+import os
+import platform
+import re
+import subprocess
+from pathlib import Path
+import tempfile
+
+def get_java_executable(known_args):
+	if known_args.java_executable is not None:
+		java_executable = known_args.java_executable
+	else:
+		java_executable = 'java'
+
+		if 'JAVA_HOME' in os.environ:
+			ext = '.exe' if platform.system()  == 'Windows' else ''
+			java_exec_to_test = Path(os.environ['JAVA_HOME']) / 'bin' / f'java{ext}'
+			if java_exec_to_test.is_file():
+				java_executable = java_exec_to_test.resolve()
+
+	if not known_args.validate_java_version:
+		return java_executable
+
+	out = subprocess.check_output([java_executable, '-version'], stderr = subprocess.STDOUT, universal_newlines=True)
+
+	matches = re.finditer(r"(?<=version\s\")(?P<major>\d+)(\.\d+\.\d+(_\d+)?)?", out)
+	for match in matches:
+		java_major_version = int(match.group("major"))
+
+		if java_major_version < 17:
+			raise Exception("jdtls requires at least Java 17")
+
+		return java_executable
+
+	raise Exception("Could not determine Java version")
+
+def find_equinox_launcher(jdtls_base_directory):
+	plugins_dir = jdtls_base_directory / "plugins"
+	launchers = plugins_dir.glob('org.eclipse.equinox.launcher_*.jar')
+	for launcher in launchers:
+		return plugins_dir / launcher
+
+	raise Exception("Cannot find equinox launcher")
+
+def get_shared_config_path(jdtls_base_path):
+	system = platform.system()
+
+	if system in ['Linux', 'FreeBSD']:
+		config_dir = 'config_linux'
+	elif system == 'Darwin':
+		config_dir = 'config_mac'
+	elif system == 'Windows':
+		config_dir = 'config_win'
+	else:
+		raise Exception("Unknown platform {} detected".format(system))
```

Comment:
Suggestion : 
raise Exception(f"Unknown platform {system} detected")

Reply:
This is part of the fixtures, this is not our code.